media-proxy/src/libfabric_ep.c

```diff
@@ -138,14 +139,28 @@ int ep_reg_mr(ep_ctx_t *ep_ctx, void *data_buf, size_t data_buf_size)
     return ret;
 }
 
-int ep_send_buf(ep_ctx_t *ep_ctx, void *buf, size_t buf_size)
-{
+int ep_send_buf(ep_ctx_t* ep_ctx, void* buf, size_t buf_size) {
+    if (!ep_ctx || !buf || buf_size == 0) {
+        ERROR("Invalid parameters provided to ep_send_buf: ep_ctx=%p, buf=%p, buf_size=%zu",
+              ep_ctx, buf, buf_size);
+        return -EINVAL;
+    }
+
     int ret;
 
     do {
-        ret = fi_send(ep_ctx->ep, buf, buf_size, ep_ctx->data_desc, ep_ctx->dest_av_entry, NULL);
-        if (ret == -EAGAIN)
-            (void)fi_cq_read(ep_ctx->cq_ctx.cq, NULL, 0);
+         // Check if the stop flag is set
+        if (ep_ctx->stop_flag) {
```

Comment:
As you added this flag my assumption is one thread sets it to let know an other thread that sending is unavailable. 
In this case volatile bool without any mutex seems insufficient to ensure that fi_send won't be called after `stop_flag` was set to true.

if this is done by single thread I think we have to allow to call on_receive only when the connection is in appropriate state. No flag would be needed then.

Reply:
Flag is needed for a gracefull shutdown, if device was replaying with -EAGAIN constantly (when on the other end nothing was received), the thread was working endlessly.